lib/puppet/provider/package/xbps.rb

```diff
@@ -0,0 +1,128 @@
+require_relative "../../../puppet/provider/package"
+require "set"
+require "uri"
+
+Puppet::Type.type(:package).provide :xbps, :parent => Puppet::Provider::Package do
+  desc "Support for the Package Manager Utility (xbps) used in VoidLinux.
+
+  This provider supports the `install_options` attribute, which allows command-line flags to be passed to xbps-install.
+  These options should be specified as an array where each element is either a string or a hash."
+
+  commands :xbps_install => "/usr/bin/xbps-install"
+  commands :xbps_remove => "/usr/bin/xbps-remove"
+  commands :xbps_query => "/usr/bin/xbps-query"
+  commands :xbps_pkgdb => "/usr/bin/xbps-pkgdb"
+
+  confine :operatingsystem => [:void]
+  defaultfor :operatingsystem => [:void]
+  has_feature :install_options, :uninstall_options, :upgradeable, :holdable, :virtual_packages
```

Comment:
I'm not familiar with `void` but is it possible to install a specific package version? If so, then you'll want `:versionable` too: https://github.com/puppetlabs/puppet/blob/7ca202bd2faf876e1904a8a2e891c10c2cee6ffc/lib/puppet/type/package.rb#L51-L54

Reply:
Well, not actually. Its a rolling release just like arch family and repository keeps cleaning itself as new version releases. They do make sure stability among packages and update revisions on same version if necessary but it doesn't let you select the older version.

Older version remains in void-packages repository tho, which can be built on the machine from source but that's a completely different matter as binary package repository isn't involved.